module/zfs/vdev.c

```diff
@@ -1167,12 +1170,17 @@ vdev_open_children(vdev_t *vd)
 	}
 	tq = taskq_create("vdev_open", children, minclsyspri,
 	    children, children, TASKQ_PREPOPULATE);
-
-	for (c = 0; c < children; c++)
-		VERIFY(taskq_dispatch(tq, vdev_open_child, vd->vdev_child[c],
-		    TQ_SLEEP) != 0);
-
-	taskq_destroy(tq);
+    
+    if(tq != NULL) { // managed to spawn thread(s)?
+        for (c = 0; c < children; c++)
+            VERIFY(taskq_dispatch(tq, vdev_open_child, vd->vdev_child[c],
+                TQ_SLEEP) != 0);
+
+        taskq_destroy(tq);
+    } else { // failed to spawn threads? Do it in the current one
+        for (c = 0; c < children; c++)
+            vdev_open_child(vd->vdev_child[c]);
+    }
```

Comment:
Functionally this looks good but you'll need to address the style issues flagged by the buildbot.  You can run the style checker locally with `make cstyle`.